### PR TITLE
feat: add proposal simulator interface function and tests

### PR DESCRIPTION
**Pull request by charleenfei**
## Description

closes: #3151

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [ ] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#pull-request-targeting)).
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/main/docs/docs/building-modules/11-structure.md) and [Go style guide](../docs/dev/go-style-guide.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/ibc-go/blob/main/testing/README.md#ibc-testing-package).
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`).
- [ ] Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Provide a [commit message](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#commit-messages) to be used for the changelog entry in the PR description for review.
- [ ] Re-reviewed `Files changed` in the Github PR explorer.
- [ ] Review `Codecov Report` in the comment section below once CI passes.

### CODE REVIEW

File: modules/apps/27-interchain-accounts/simulation/proposals.go [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+)
+
+// Simulation operation weights constants
+const (
+	DefaultWeightMsgUpdateParams int = 100
+	OpWeightMsgUpdateParams          = "op_weight_msg_update_params" // #nosec
+)
+
+// ProposalMsgs defines the module weighted proposals' contents
+func ProposalMsgs() []simtypes.WeightedProposalMsg {
```
[the remark this answers is not in the document]
Comment (charleenfei) [reply modules/apps/27-interchain-accounts/simulation/proposals.go:21]: you're absolutely correct! i just wanted to verify these were the right/only places to impl these functions, i wasn't entirely sure if i was missing any app modules.

**Review by damiannolan — COMMENTED** [review]
Nice! Looks good, do we need to add these for transfer? or they are already there?

File: modules/apps/27-interchain-accounts/simulation/proposals.go [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+	simtypes "github.com/cosmos/cosmos-sdk/types/simulation"
+	"github.com/cosmos/cosmos-sdk/x/simulation"
+
+	"github.com/cosmos/ibc-go/v7/modules/apps/27-interchain-accounts/host/types"
+)
+
+// Simulation operation weights constants
+const (
+	DefaultWeightMsgUpdateParams int = 100
+	OpWeightMsgUpdateParams          = "op_weight_msg_update_params" // #nosec
```
[the remark this answers is not in the document]
Comment (damiannolan) [reply modules/apps/27-interchain-accounts/simulation/proposals.go]: that's obnoxious af

**Comment by charleenfei** [comment]
aha, i forgot about the `UpdateParams` gov-gated transfer message. will update this PR!

File: modules/apps/27-interchain-accounts/simulation/proposals.go [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+)
+
+// ProposalMsgs defines the module weighted proposals' contents
+func ProposalMsgs() []simtypes.WeightedProposalMsg {
+	return []simtypes.WeightedProposalMsg{
+		simulation.NewWeightedProposalMsg(
+			OpWeightMsgUpdateParams,
+			DefaultWeightMsgUpdateParams,
+			SimulateMsgUpdateParams,
+		),
```
Comment (crodriguezvega) [inline modules/apps/27-interchain-accounts/simulation/proposals.go:28]: Should we add a second element to the slice with [`MsgUpdateParams` for ICA controller](https://github.com/cosmos/ibc-go/blob/5ca37ef6e56a98683cf2b3b1570619dc9b322977/proto/ibc/applications/interchain_accounts/controller/v1/tx.proto#L65)?

File: modules/core/simulation/proposals.go [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+)
+
+// ProposalMsgs defines the module weighted proposals' contents
+func ProposalMsgs() []simtypes.WeightedProposalMsg {
+	return []simtypes.WeightedProposalMsg{
+		simulation.NewWeightedProposalMsg(
+			OpWeightMsgUpdateParams,
+			DefaultWeightMsgUpdateParams,
+			SimulateMsgUpdateParams,
+		),
```
Comment (crodriguezvega) [inline modules/core/simulation/proposals.go:28]: Same comment about whether we should add a second element to the slice using [`MsgUpdateParams` for 03-connection](https://github.com/cosmos/ibc-go/blob/5ca37ef6e56a98683cf2b3b1570619dc9b322977/proto/ibc/core/connection/v1/tx.proto#L131)?

**Review by crodriguezvega — COMMENTED** [review]
Thanks, @charleenfei. Left a couple of questions.❓

File: modules/apps/27-interchain-accounts/simulation/proposals.go [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+	}
+}
+
+// SimulateMsgUpdateParams returns a random MsgUpdateParams
+func SimulateMsgUpdateParams(r *rand.Rand, _ sdk.Context, _ []simtypes.Account) sdk.Msg {
+	var authority sdk.AccAddress = address.Module("gov")
+	params := types.DefaultParams()
+	params.HostEnabled = false
```
[the remark this answers is not in the document]
Comment (charleenfei) [reply modules/apps/27-interchain-accounts/simulation/proposals.go:36]: sure, we can do this!

[the remark this answers is not in the document]
Comment (charleenfei) [reply modules/apps/27-interchain-accounts/simulation/proposals.go:36]: opened [this tracking issue](https://github.com/cosmos/ibc-go/issues/4595)

**Review by crodriguezvega — APPROVED** [review]
Nice work. Thank you, @charleenfei.

File: modules/apps/27-interchain-accounts/simulation/proposals.go (hunk 5 above)
[the remark this answers is not in the document]
Comment (colin-axner) [reply modules/apps/27-interchain-accounts/simulation/proposals.go:42]: I think the purpose of simulations is that it performs random operations, we don't utilize simulations so this is mostly for chains which run sdk simulations

**Review by colin-axner — APPROVED** [review]
LGTM, thanks @charleenfei

**Comment by colin-axner** [comment]
this can target main? unless you are planning on adding the new proposal msg types as well?

**Pull request merged**